pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -317,10 +317,10 @@ func (plugin *kubenetNetworkPlugin) SetUpPod(namespace string, name string, id k
 	if err != nil {
 		return err
 	}
-	if res.IP4 == nil || res.IP4.IP.String() == "" {
+	if res.IP4 == nil || res.IP4.IP.IP.String() == "" {
```

Comment:
You're right, it returns either <nil> or ? for invalid addresses.  Looks like the correct check should actually be len(res.IP4.IP.IP) == IPv4len.